### PR TITLE
rustc_codegen_llvm: less pubs

### PR DESCRIPTION
**Pull request by klensy**
This removes few pubs around and slaps `dead_code`s around.

### CODE REVIEW

File: compiler/rustc_codegen_llvm/src/llvm/ffi.rs [hunk 1]
```diff
@@ -289,6 +295,7 @@ impl RealPredicate {
 /// LLVMTypeKind
 #[derive(Copy, Clone, PartialEq, Debug)]
 #[repr(C)]
+#[allow(dead_code)]
 pub enum TypeKind {
```
Comment (klensy) [inline compiler/rustc_codegen_llvm/src/llvm/ffi.rs:299]: This suspiciously unused, `TypeKind` from `rustc_codegen_ssa` used around instead.

**Comment by tgross35** [comment]
Could we use `#[expect(...)]` instead of `allow` since that is available now? So we remember to remove the attributes if they wind up being used

**Comment by cjgillot** [comment]
Why not just remove the dead code?

**Comment by klensy** [comment]
> Why not just remove the dead code?

Tail call stuff was added in #112791 and in process of implementation, as i understand; other ones require changes in c++ part, will check that.

> Could we use #[expect(...)] instead of allow

Nice, will change that stuff.

File: compiler/rustc_codegen_llvm/src/llvm/ffi.rs (hunk 1 above)
Comment (klensy) [reply compiler/rustc_codegen_llvm/src/llvm/ffi.rs:299]: This is #85677

**Comment by JohnCSimon** [comment]
@klensy 
ping from triage - can you post your status on this PR? This PR has not received an update in a few months.

**Comment by alex-semenyuk** [comment]
@klensy
Form wg-triage. Closed this PR due to inactivity. Feel free to reopen or raised new one. Thanks for your efforts.

**Pull request closed without merging**